src/BundleAdjustmentNLSFunctions.jl

```diff
@@ -46,18 +46,26 @@ mutable struct BundleAdjustmentModel{T, S} <: AbstractNLSModel{T, S}
   P2_vec::S
 end
 
-""" Create name from filename """
-function name(filename::AbstractString)
-  return splitdir(splitext(splitext(filename)[1])[1])[2]
-end
-
 """
-    BundleAdjustmentModel(filename::AbstractString; T::Type=Float64, verbose::Bool=false)
+    BundleAdjustmentModel(name::AbstractString; T::Type=Float64, direct_path::Bool = False)
 
-Constructor of BundleAdjustmentModel, creates an NLSModel from a BundleAdjustment archive
+Constructor of BundleAdjustmentModel, creates an NLSModel with name `name` from a BundleAdjustment archive with precision `T`.
+Set `direct_path` to `True` if name is the path to the archive itself instead of the name of the problem.
```

Comment:
It's `true` in Julia. I don't understand what `direct_path` does. You should be able to detect if it's a path or not.

Reply:
I didn't think about it I will do it automatically.